discord/ext/commands/core.py

```diff
@@ -160,7 +204,7 @@ def pop(self, k, default=None):
     def __setitem__(self, k, v):
         super().__setitem__(k.casefold(), v)
 
-class Command(_BaseCommand):
+class Command(_BaseCommand, Generic[CT, P, T]):
```

Comment:
Do we like ParamSpec?

Pros:
 - command callable arguments will be validated in-decorators.
 - `Command.__call__` / `Context.invoke` will validate arguments.

Cons:
 - code is unreadable / requires arcane knowledge to work with

Reply:
I think the code is pretty well readable and with better names for `TypeVar`s, the code will be come much more readable